### PR TITLE
Add throughput limiting when flushing data to disk

### PR DESCRIPTION
**Pull request by xichen2020**
cc @robskillington @martin-mao

### CODE REVIEW

File: persist/fs/manager.go [hunk 1]
```diff
@@ -23,15 +23,34 @@ package fs
...
+// persistManager is responsible for persisting series segments onto local filesystem.
+// It is not thread-safe.
 type persistManager struct {
-	opts           Options
-	filePathPrefix string
-	writer         FileSetWriter
+	opts                       Options
+	filePathPrefix             string
+	throughputCheckInterval    time.Duration
+	throughputLimitMBPerSecond float64
```
Comment (xichen2020) [inline persist/fs/manager.go]: Sure yeah

File: storage/flush.go [hunk 2]
```diff
@@ -25,34 +25,47 @@ import (
...
+		nowFn:       opts.ClockOptions().NowFn(),
 		blockSize:   blockSize,
 		pm:          pm,
 		flushStates: map[time.Time]fileOpState{},
+		metrics:     instrument.NewMethodMetrics(scope, "flush", 1.0),
 	}
 }
 
+func (m *flushManager) IsFlushing() bool {
+	return m.flushInProgress
```
Comment (xichen2020) [inline storage/flush.go]: Good catch

File: persist/fs/manager.go [hunk 3]
```diff
@@ -23,15 +23,34 @@ package fs
...
 	"time"
 
+	"github.com/m3db/m3db/clock"
 	"github.com/m3db/m3db/persist"
 	"github.com/m3db/m3db/ts"
 )
 
-// persistManager is responsible for persisting series segments onto local filesystem
+const (
+	bytesPerMb = 1024 * 1024 / 8
```
Comment (robskillington) [inline persist/fs/manager.go]: Perhaps to be super clear might be best to call this `bytesPerMegabit`?

File: persist/fs/manager.go [hunk 4]
```diff
@@ -23,15 +23,34 @@ package fs
...
+)
+
+var (
+	zeroTime time.Time
```
Comment (robskillington) [inline persist/fs/manager.go]: Every other package we've named this `timeZero` I believe.  Perhaps we can use this name instead for consistency?

File: persist/fs/types.go [hunk 5]
```diff
@@ -111,4 +118,16 @@ type Options interface {
...
 	ReaderBufferSize() int
+
+	// SetThroughputCheckInterval sets the filesystem throughput check interval
+	SetThroughputCheckInterval(value time.Duration) Options
+
+	// ThroughputCheckInterval returns the filesystem throughput check interval
+	ThroughputCheckInterval() time.Duration
+
+	// SetThroughutLimitMbps sets the filesystem throughput limit
+	SetThroughutLimitMbps(value float64) Options
```
Comment (robskillington) [inline persist/fs/types.go]: Spelling mistake here `SetThroughutLimitMbps`

File: persist/fs/types.go [hunk 6]
```diff
@@ -111,4 +118,16 @@ type Options interface {
...
+
+	// ThroughutLimitMbps returns the filesystem throughput limit
+	ThroughutLimitMbps() float64
```
Comment (robskillington) [inline persist/fs/types.go]: And here.

**Pull request merged**